README.md
# react-webpack

> 编译上线使用命令
`npm run build`
或
`yarn build`

> 开发使用命令
`npm run dev`
或
`yarn dev`

# 更新日志

### 2019-01-08
- 升级 react react-router webpack


CHANGELOG.md
# 1.0.0 (2020-02-25)


### Features

* **增加git-cz组件用于替代git commit提交的内容标准化，可生成changelog:** 增加git-cz组件用于替代git commit提交的内容标准化，可生成changelog ([2098f93](https://github.com/zhangjie1314/react-webpack/commit/2098f93f7ffa90246444f5cfe4ca23bd6eb898c9))
* 123123123123 ([7847127](https://github.com/zhangjie1314/react-webpack/commit/78471278a0a275b0b772ba11f38d6fccd8269986))



# 1.0.0 (2020-02-25)


### Features

* **增加git-cz组件用于替代git commit提交的内容标准化，可生成changelog:** 增加git-cz组件用于替代git commit提交的内容标准化，可生成changelog ([2098f93](https://github.com/zhangjie1314/react-webpack/commit/2098f93f7ffa90246444f5cfe4ca23bd6eb898c9))
* 123123123123 ([7847127](https://github.com/zhangjie1314/react-webpack/commit/78471278a0a275b0b772ba11f38d6fccd8269986))





webpack.production.config.js
var path = require('path')

// 文件路径
var ROOT_PATH = path.resolve(__dirname)
var APP_PATH = path.resolve(ROOT_PATH, 'src')
var BUILD_PATH = path.resolve(ROOT_PATH, 'build')
var TEM_PATH = path.resolve(ROOT_PATH, 'templates')

// 开发浏览器端口
const port = 10010

// 插件
var HtmlWebpackPlugin = require('html-webpack-plugin'),
    HWPlugin = new HtmlWebpackPlugin({
        template: path.resolve(TEM_PATH, 'index.html'),
        filename: 'index.html',
        chunks: [ 'app', 'runtime~app', 'vendors-app' ],
        inject: 'body',
        minify: {
            //压缩HTML文件
            removeComments: true, //移除HTML中的注释
            collapseWhitespace: true //删除空白符与换行符
        }
    })
var CleanWebpackPlugin = require('clean-webpack-plugin'),
    CWPlugin = new CleanWebpackPlugin()

var configs = {
    // 可选值 none, development , production
    mode: 'production',
    entry: {
        app: [ path.resolve(APP_PATH, 'app.jsx') ]
    },
    output: {
        path: BUILD_PATH,
        filename: 'js/[name].js'
    },
    module: {
        rules: [
            {
                test: /.(jsx|js|mjs)$/,
                enforce: 'pre',
                exclude: /node_modules/,
                loader: 'eslint-loader',
                options: {
                    emitError: true,
                    emitWarning: true
                }
            },
            {
                test: /\.(jsx|js|mjs)$/,
                include: APP_PATH,
                loader: 'babel-loader',
                options: {
                    presets: [
                        '@babel/preset-env',
                        '@babel/preset-react',
                        { plugins: [ '@babel/plugin-proposal-class-properties' ] }
                    ]
                }
            },
            {
                test: /\.less$/,
                loader: 'style-loader!css-loader!less-loader'
            },
            {
                test: /\.css$/,
                loader: 'style-loader!css-loader'
            },
            {
                test: /\.(png|jpg|woff|svg|eot|ttf)\??.*$/,
                loader: 'url-loader?limit=8192&name=img/[name].[hash:8].[ext]'
            }
        ]
    },
    resolve: {
        // require文件时省略文件的扩展名
        extensions: [ '.js', '.jsx', '.json' ]
    },
    optimization: {
        runtimeChunk: true,
        splitChunks: {
            // 有效值为all，async和initial
            chunks: 'all',
            // 指定名称分隔符
            automaticNameDelimiter: '-'
        }
    },
    plugins: [ CWPlugin, HWPlugin ]
}

module.exports = configs


webpack.config.js
var path = require('path')
var webpack = require('webpack')
const { CleanWebpackPlugin } = require('clean-webpack-plugin')

// 文件路径
var ROOT_PATH = path.resolve(__dirname)
var APP_PATH = path.resolve(ROOT_PATH, 'src')
var BUILD_PATH = path.resolve(ROOT_PATH, 'build')
var TEM_PATH = path.resolve(ROOT_PATH, 'templates')

// 开发浏览器端口
const port = 10010

// 插件
var HtmlWebpackPlugin = require('html-webpack-plugin'),
    HWPlugin = new HtmlWebpackPlugin({
        template: path.resolve(TEM_PATH, 'index.html'),
        filename: 'index.html',
        chunks: [ 'app', 'runtime~app', 'vendors-app' ],
        inject: 'body',
        minify: {
            //压缩HTML文件
            removeComments: true, //移除HTML中的注释
            collapseWhitespace: true //删除空白符与换行符
        }
    })

var configs = {
    // 可选值 none, development , production
    mode: 'development',
    entry: {
        app: [ path.resolve(APP_PATH, 'app.jsx') ]
    },
    output: {
        path: BUILD_PATH,
        filename: 'js/[name].js'
    },
    // 启动dev source map，出错以后就会采用source-map的形式直接显示你出错代码的位置
    devtool: 'source-map',
    // 打开开发服务
    devServer: {
        historyApiFallback: true, //spa不跳转,history模式的路由需要true
        compress: true,
        host: 'localhost',
        port: port,
        hot: true,
        inline: true,
        open: true,
        contentBase: BUILD_PATH,
        progress: true
    },
    module: {
        rules: [
            {
                test: /.(jsx|js|mjs)$/,
                enforce: 'pre',
                exclude: /node_modules/,
                loader: 'eslint-loader',
                options: {
                    emitError: true,
                    emitWarning: true
                }
            },
            {
                test: /\.(jsx|js|mjs)$/,
                include: APP_PATH,
                loader: 'babel-loader',
                options: {
                    presets: [
                        '@babel/preset-env',
                        '@babel/preset-react',
                        { plugins: [ '@babel/plugin-proposal-class-properties' ] }
                    ]
                }
            },
            {
                test: /\.less$/,
                loader: 'style-loader!css-loader!less-loader'
            },
            {
                test: /\.css$/,
                loader: 'style-loader!css-loader'
            },
            {
                test: /\.(png|jpg|woff|svg|eot|ttf)\??.*$/,
                loader: 'url-loader?limit=8192&name=img/[name].[hash:8].[ext]'
            }
        ]
    },
    resolve: {
        // require文件时省略文件的扩展名
        extensions: [ '.js', '.jsx', '.json' ]
    },
    optimization: {
        runtimeChunk: true,
        splitChunks: {
            // 有效值为all，async和initial
            chunks: 'all',
            // 指定名称分隔符
            automaticNameDelimiter: '-'
        }
    },
    plugins: [ new CleanWebpackPlugin(), HWPlugin, new webpack.HotModuleReplacementPlugin() ]
}

module.exports = configs
